autonomy/deploy/base.py

```diff
@@ -186,12 +186,6 @@ def verify_agent_instances(
                 f"Key file contains keys which are not registered as instances; invalid keys={key_not_in_instances}"
             )
 
-        instances_not_in_keys = instances.difference(addresses)
-        if instances_not_in_keys:
-            raise NotValidKeysFile(
-                f"Key file does not contain key pair for following instances {instances_not_in_keys}"
-            )
```

Comment:
We should not raise if the keys file does not contain key pair for every single registered agent instance

Reply:
it was here for a reason though. Should we log the addresses for which we do and do not have keys?

Reply:
We should indeed log something. Otherwise it's hard to use this magical tool. How about logging the detected addresses and their matching instance identifiers? This would be helpful in the logs

Reply:
[d296dc9](https://github.com/valory-xyz/open-autonomy/pull/1612/commits/d296dc9f56d6f83cb8806997ac98ff426b092424)